packages/wdio-browserstack-service/src/service.js

```diff
@@ -0,0 +1,131 @@
+import logger from '@wdio/logger'
+import request from 'request'
+
+const log = logger('wdio-browserstack-service');
+
+export default class BrowserstackService {
+    constructor (config) {
+        this.config = config
+    }
+
+    before() {
+        this.sessionId = global.browser.sessionId;
+        this.failures = 0;
+        this.auth = global.browser.requestHandler ? global.browser.requestHandler.auth || {} : {};
```

Comment:
let's define these 3 variables in the constructor too

Reply:
Let's do 
```
this.auth = (global.browser.requestHandler && global.browser.requestHandler.auth) || {};
```

Reply:
Your first request isn't completely doable. When i moved the sessionID and auth to the constructor, the test wouldn't start. I still moved the failures property up though.

I fixed the auth assignment as requested.